File: code35.py
class Solution(object):
    def searchInsert(self, nums, target):
        startIndex = self.firstIndex(nums,target,0,len(nums)-1)
        return startIndex
    def firstIndex(self,nums,target,lo,hi):
        if lo==hi and (nums[lo]==target or nums[lo]>target) or lo>hi:
            return lo
        elif lo==hi:
            return hi+1
        mid=int((lo+hi)/2)
        if target>nums[mid]:
            return self.firstIndex(nums,target,mid+1,hi)
        else:
            return self.firstIndex(nums,target,lo,mid)
s=Solution()
nums=[1,3,5,6]
print(s.searchInsert(nums,5))
print(s.searchInsert(nums,2))
print(s.searchInsert(nums,7))
print(s.searchInsert(nums,0))
print(s.searchInsert(nums,6))
lst=[]
print(s.searchInsert(lst,11))

File: code35.cpp

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int left = 0;
        int right = nums.size()-1;
        while (left < right)
        {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target)
                return mid;
            if (nums[mid] < target)
                left = mid + 1;
            else
                right = mid - 1;
        }
        if (target <= nums[left])
            return left;
        else
            return left + 1;
    }
};

int main()
{
    vector<int> i1 = {1, 3, 5, 6};
    Solution s;
    cout<<s.searchInsert(i1,5)<<endl;
    cout<<s.searchInsert(i1,2)<<endl;
    cout<<s.searchInsert(i1,4)<<endl;
    cout<<s.searchInsert(i1,7)<<endl;
}